### DISCUSSION TITLE
`Core(Device(Lost))` error when launching with nsight graphics GPU profiler

### DISCUSSION THREAD

**Discussion by SarthakSingh31** [gfx-rs/wgpu#6820] [Q&A]
**Description**
Launching the `wgpu-examples` from NVIDIA Nsight Graphics with the GPU Trace Profiler Activity leads to the application crashing due to `RequestDeviceError { inner: Core(Device(Lost)) }` error when requesting the device when forcing the vulkan backend.
```
2024-12-21T09:14:52.470204Z  INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 960", vendor: 4318, device: 5121, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "565.77", backend: Vulkan }
thread 'main' panicked at /home/ssingh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.15.0/src/renderer/mod.rs:363:10:
called `Result::unwrap()` on an `Err` value: RequestDeviceError { inner: Core(Device(Lost)) }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
When I allow it to use any backend it errors out with
```
thread 'main' panicked at /home/ssingh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/khronos-egl-6.0.0/src/lib.rs:1095:27:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: core::option::unwrap_failed
   4: <wgpu_hal::gles::egl::Instance as wgpu_hal::Instance>::enumerate_adapters
   5: <I as wgpu_hal::dynamic::instance::DynInstance>::enumerate_adapters
   6: wgpu_core::instance::Instance::request_adapter
   7: wgpu_core::instance::<impl wgpu_core::global::Global>::request_adapter
   8: <T as wgpu::context::DynContext>::instance_request_adapter
   9: wgpu::api::instance::Instance::request_adapter
  10: futures_lite::future::block_on
  11: <bevy_render::RenderPlugin as bevy_app::plugin::Plugin>::build
  12: bevy_app::app::App::add_boxed_plugin
  13: bevy_app::plugin_group::PluginGroupBuilder::finish
  14: <(S0,S1,S2) as bevy_app::plugin::sealed::Plugins<(bevy_app::plugin::sealed::PluginsTupleMarker,P0,P1,P2)>>::add_to_app
  15: bevy_app::app::App::add_plugins
  16: simple::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Repro steps**
1. Launch NVIDIA Nsight Graphics (version: 2024.3.0.0).
2. Click on start activity.
3. Fill in the application path and application workspace.
4. Select the `GPU Trace Profiler` from the activity menu on the bottom left.
5. Click on `Launch GPU Trace` button at the bottom right.

**Expected vs observed behavior**
Expected: The application runs without any issues and I can profile the shader.
Observed: The application crashes and fails to request a device.

**Platform**
Platform: Arch Linux (`Linux version 6.12.4-arch1-1 (linux@archlinux) (gcc (GCC) 14.2.1 20240910, GNU ld (GNU Binutils) 2.43.0) #1 SMP PREEMPT_DYNAMIC Mon, 09 Dec 2024 14:31:57 +0000`)
Wgpu: `main` branch
GPU: GTX 960 6GB

**Comment by SarthakSingh31** [comment]
If anyone could recommend a different shader profiler that works that would also unblock me. I just started using NVIDIA Nsight Graphics so I am not married to it.

**Comment by cwfitzgerald** [comment]
I have no idea what would cause this - but I know that nsight doesn't support profiling vulkan on 10xx or older, they only support 20xx or newer.

I think this isn't our issue.

As for a recommendation, I would look into tracy and tracy's gpu spans. You can use https://github.com/Wumpf/wgpu-profiler to hook into tracy's gpu infrastructure.
